### PR TITLE
Bug fix, add gelu for torch, add lambda lr scheduler for torch, allow `predict` to handle CartesianProd` data for torch, automatically change the dataset's datatype for torch.

### PR DESCRIPTION
**Pull request by supercgor**
1. In torch, the `predict` doesn't accept tuple input, that is a bug.
2. The second thing is for `CartesianProd`, `predict` can not deal with the differential equation `operator`. So added some code for torch to resolve it.
3. automatically change the data type for torch training.

* add torch gelu

* add lambda lr scheduler for torch

* repair missing deeponet for pytorch

### CODE REVIEW

File: deepxde/backend/backend.py [hunk 1]
```diff
@@ -274,6 +274,8 @@ def elu(x):
 def relu(x):
     """Applies the rectified linear unit activation function."""
```
Comment (lululxvi) [inline deepxde/backend/backend.py:276]: Two empty lines

**Comment by lululxvi** [comment]
Split the PR into a few independent ones.
